runtime/parser2/expression.go

```diff
@@ -428,3 +442,184 @@ func applyLeftDenotation(p *parser, tokenType lexer.TokenType, left ast.Expressi
 	}
 	return leftDenotation(p, left)
 }
+
+// parseStringLiteral parses a whole string literal, including start and end quotes
+//
+func parseStringLiteral(s string) (result string, errs []error) {
+	report := func(err error) {
+		errs = append(errs, err)
+	}
+
+	l := len(s)
+	if l == 0 {
+		report(fmt.Errorf("missing start of string literal: expected '\"'"))
+		return
+	}
+
+	if l >= 1 {
+		first := s[0]
+		if first != '"' {
+			report(fmt.Errorf("invalid start of string literal: expected '\"', got %q", first))
+		}
+	}
+
+	missingEnd := false
+	endOffset := l
+	if l >= 2 {
+		last := s[l-1]
+		if last != '"' {
+			missingEnd = true
+		} else {
+			endOffset = l - 1
+		}
+	} else {
+		missingEnd = true
+	}
+
+	var innerErrs []error
+	result, innerErrs = parseStringLiteralContent(s[1:endOffset])
+	errs = append(errs, innerErrs...)
+
+	if missingEnd {
+		report(fmt.Errorf("invalid end of string literal: missing '\"'"))
+	}
+
+	return
+}
+
+// parseStringLiteralContent parses the string literal contents, excluding start and end quotes
+//
+func parseStringLiteralContent(s string) (result string, errs []error) {
+
+	var builder strings.Builder
+	defer func() {
+		result = builder.String()
+	}()
+
+	report := func(err error) {
+		errs = append(errs, err)
+	}
+
+	l := len(s)
+
+	var r rune
+	i := 0
```

Comment:
Super minor, but might be good to name this to something a bit more descriptive

Reply:
💯 Tried to improve it in 782be24